### ISSUE TITLE
args[module] reports fails for mutually exclusive arguments although they are default(omit)

### ISSUE THREAD

**Issue by ianw** [ansible/ansible-lint#2962]
##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.11.0 using ansible 2.14.2
```

The following call

```yaml
- name: Add apt keys
  become: true
  apt_key:
    url: "{{ zj_item['url'] | default(omit) }}"
    data: "{{ zj_item['data'] | default(omit) }}"
  loop: "{{ repositories_keys }}"
  loop_control:
    loop_var: zj_item
```

reports

```
roles/ensure-package-repositories/tasks/Debian.yaml:8: args[module]: parameters are mutually exclusive: data|file|keyserver|url (warning)
```

I guess it is true that the arguments are mutually exclusive, but also because they're ```default(omit)``` only one is coming in, making it valid code.

Feels like a bit of a tricky one, because the code could be correct, but it would be hard to determine statically.  I guess you could split it up into multiple calls behind ```when:``` statements, but at the cost of extra task overheads.  Perhaps ```default(omit)``` arguments should be ignored for mutually-exclusive argument checking?

**Comment by cidrblock** [comment]
Since the value of the jinja expressions are only available during the playbook run and could vary between multiple runs of the playbook, it is really difficult if not impossible to lint this one, in this case it might be best to `noqa` the task https://ansible-lint.readthedocs.io/rules/args/#special-cases

The other option might be to do a conditionally include, which could help catch the case if both were defined or neither (totally untested)
```
- name: Run the appropriate task
  include_tasks: use-{{ value }}.yaml
  vars:
    value: >-
    {%- if 'arg' in zj_item and not 'data' in zj_item -%}
    'arg'
    {%- elif 'data' in zj_item and 'arg' not in zj_item -%}
    'data'
    {%- else -%}
    'fail'
    {%- end %-} 
  loop_control:
    loop_var: zj_item
```
and then have 3 files, use-arg.yaml, use-data.yaml, and use-fail.yaml to run the appropriate task

**Comment by klaus-tux** [comment]
At the moment the implementation of the `args[module]` rule suppresses two kinds of errors if the value contains a jinja expression:

- If there is a TypeError (`"argument '<name>' is of type"`)
- If the given value not in the list of allowed choices (`"value of <name> must be one of:"`)

(The rule implementation does not look into the jinja expression itself.)

Contradictory, this special case (mutually exclusive values contain jinja expressions) is not handled by the rule, but the rule documentation states, that the user can suppress it with `# noqa: args[module]`.

This seems inconsistent to me.

I see the following options:

1. Suppress the mutually exclusive error if there is a jinja expression. In this case the documentation should be updated with a list of suppressed errors in case of jinja templates.
2. Remove the suppression of the TypeError and the value not in choices case. In this case the documentation should be updated with examples to use `# noqa: args[module]` in these cases.

Any comments or suggestions on this?
